pymc3/distributions/logp.py

```diff
@@ -260,6 +267,86 @@ def _logp(
     """
     value_var = rvs_to_values.get(var, var)
     return at.zeros_like(value_var)
+    # raise NotImplementedError(f"Logp cannot be computed for op {op}")
+
+
+@_logp.register(Elemwise)
+def elemwise_logp(op, *args, **kwargs):
+    return _logp(op.scalar_op, *args, **kwargs)
+
+
+# TODO: Implement DimShuffle logp?
+# @_logp.register(DimShuffle)
+# def logp_dimshuffle(op, var, *args, **kwargs):
+#     if var.owner and len(var.owner.inputs) == 1:
+#         inp = var.owner.inputs[0]
+#         if inp.owner and hasattr(inp.owner, 'op'):
+#             return _logp(inp.owner.op, inp, *args, **kwargs)
+#     raise NotImplementedError
+
+
+@_logp.register(Add)
+@_logp.register(Mul)
+def linear_logp(op, var, rvs_to_values, *linear_inputs, **kwargs):
+
+    if len(linear_inputs) != 2:
+        raise ValueError(f"Expected 2 inputs but got: {len(linear_inputs)}")
+
+    # Find base_rv and constant inputs
+    base_rv = []
+    constant = []
+    for inp in linear_inputs:
+        res_ancestors = list(walk_model((inp,), walk_past_rvs=True))
+        # unregistered variables do not contain a value_var tag
+        res_unregistered_ancestors = [
+            v
+            for v in res_ancestors
+            if v.owner
+            and isinstance(v.owner.op, RandomVariable)
+            and not getattr(v.tag, "value_var", False)
+        ]
+        if res_unregistered_ancestors:
+            base_rv.append(inp)
+        else:
+            constant.append(inp)
+
+    if len(base_rv) != 1:
+        raise NotImplementedError(
+            f"Logp of linear transform requires one branch with an unregistered RandomVariable but got {len(base_rv)}"
+        )
+
+    base_rv = base_rv[0]
+    constant = constant[0]
+    var_value = rvs_to_values.get(var, var)
+
+    # Get logp of base_rv
+    base_value = base_rv.type()
+    logp_base_rv = logpt(base_rv, {base_rv: base_value}, **kwargs)
+    fgraph = FunctionGraph(
+        [i for i in graph_inputs((logp_base_rv,)) if not isinstance(i, Constant)],
+        outputs=[logp_base_rv],
+        clone=False,
+    )
+
+    # Transform base_rv and apply jacobian correction (for continuous rvs)
+    if isinstance(op, Add):
+        fgraph.replace(base_value, var_value - constant, import_missing=True)
+        logp_linear_rv = fgraph.outputs[0]
+    elif isinstance(op, Mul):
+        fgraph.replace(base_value, var_value / constant, import_missing=True)
+        logp_linear_rv = fgraph.outputs[0]
+        if "float" in base_rv.dtype:
+            logp_linear_rv -= at.log(at.abs_(constant))
+
+    # Replace rvs in graph
+    (logp_linear_rv,), _ = rvs_to_value_vars(
+        (logp_linear_rv,),
+        apply_transforms=kwargs.get("transformed", True),
```

Comment:
This was just temporarily. I had the gut feeling this replacement shouldn't be inside this method to begin with, but I don't have a better place to put it. Feels like things would be a bit more clean if we separated the rewrite in two steps:
1. Introduce all logp terms in the graph (keeping the rvs as inputs)
2. Replace all rvs by value_vars (and only now apply transforms)

My confidence that this would be helpful is not much though :P